docs/10_dart_code_style.md

```diff
@@ -0,0 +1,100 @@
+# Dart code style
+> Summary of the proposed change
+
+Make sure our code style is consistent in parts that are not covered in the Dart code style.
+
+# References
+> Link to supporting documentation, GitHub tickets, etc.
+
+- [Effective Dart](https://dart.dev/guides/language/effective-dart/) is the main source for 
+all code style, documentation, design and usage conventions.
+
+
+# Motivation
+> What problem is this project solving?
+
+As Dart has a great code style available
+that we use on this project we noticed that in some real-world cases there is no guidance,
+so here we collect all cases that we think important to clarify to make sure that the code style
+is consistent through the project.
+
+# Goals
+
+> Identify success metrics and measurable goals.
+
+Have a document that can be used in code reviews to clarify code style issues and make sure
+that code style is consistent.
+
+# Non-Goals
+
+> Identify what's not in scope.
+
+- Other languages we might use (Bash scripts, JavaScript).
+
+# Documentation
+
+## Unit tests
+1. Test Group descriptions start with Capital letter:
+    ```dart
+    group("DashboardPage", () {
+    ```
+2. Test descriptions inside Test Groups start with a small letter:
+    ```dart
+    test("should throw ArgumentError trying to create an instance with null CI client", 
+    ```
+3. Test descriptions not inside Test Groups start with a Capital letter:
+    ```dart
+    testWidgets("Can't create widget without data",
+   ```
+4. Group and test descriptions should not end with a dot (.).
+5. If a test is specific for a method (or named constructor) - test description should start with 
+a dot (.) followed by a method name with an empty parenthesis (without parameters even if there are any):
+
+    ```dart
+    test(".fromJson() should create an instance from the json map", () {
+    ```
+6. Use Test Group descriptions with class under test name:
+    ```dart
+    group("ReceiveProjectMetricUpdates", () {
+       .....
+       test("loads all fields in the performance metrics", () {
+    ```
+7. Multiline test descriptions shouldn't be used as such tests can't 
+be started in [Intellij](https://youtrack.jetbrains.com/issue/WEB-44842).
+8. Prefer using double quotes. 
```

Comment:
Maybe we should add examples for these points as for others above? We can use the same style as the Effective Dart do. I.e., provide `bad` and `good` examples and use `DO`, `DO NOT`, etc. words.

Reply:
Thanks @vladlaba - please check the new version and let me know what you think 💭